                                        DEVASHISH NALAPAREDDY - 1002162185
                                        Haripranay Peddagolla- 1002168970
                                        Roja Bobbala -         1002170073






# Assignment 3 - Association Rule Mining

## Instructions

For the Third Assignment, we are going to be performing association rule mining with Apriori and FP-Growth algorithm.

1. There are set of questions associated with each task for apriori and fp-growth respectively.
2. The dataset used for whole assignment is same.
3. Each Task is presenting a challenge.

Data variable Descriptions:

- InvoiceNo: Invoice Number -> If this code starts with C, it means that the operation has been canceled.
- StockCode: Product Code -> Unique number for each product
- Description: Product name
- Quantity: Number of products -> how many of the products on the invoices were sold.
- InvoiceDate
- UnitePrice
- CustomerID: Unique customer number
- Country

Simply run the blow cell to load the dataset.

In [406]:
import pandas as pd

data = pd.read_csv("Data_batch2.csv", encoding= 'unicode_escape')
data.head()

Invoice StockCode                          Description  Quantity  \
0  536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1  536365     71053                  WHITE METAL LANTERN         6   
2  536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3  536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4  536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   

      InvoiceDate  Price  Customer ID         Country  
0  12/1/2010 8:26   2.55      17850.0  United Kingdom  
1  12/1/2010 8:26   3.39      17850.0  United Kingdom  
2  12/1/2010 8:26   2.75      17850.0  United Kingdom  
3  12/1/2010 8:26   3.39      17850.0  United Kingdom  
4  12/1/2010 8:26   3.39      17850.0  United Kingdom

### Task A: 2 points

1. Print data details
2. Check for null values and solve them using best solution.

In [407]:
display(data.info())
display(data.describe())



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541910 entries, 0 to 541909
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   Invoice      541910 non-null  object 
 1   StockCode    541910 non-null  object 
 2   Description  540456 non-null  object 
 3   Quantity     541910 non-null  int64  
 4   InvoiceDate  541910 non-null  object 
 5   Price        541910 non-null  float64
 6   Customer ID  406830 non-null  float64
 7   Country      541910 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 33.1+ MB


None

Quantity          Price    Customer ID
count  541910.000000  541910.000000  406830.000000
mean        9.552234       4.611138   15287.684160
std       218.080957      96.759765    1713.603074
min    -80995.000000  -11062.060000   12346.000000
25%         1.000000       1.250000   13953.000000
50%         3.000000       2.080000   15152.000000
75%        10.000000       4.130000   16791.000000
max     80995.000000   38970.000000   18287.000000

In [408]:
# Checking for null values
display(data.isnull().sum())



Invoice             0
StockCode           0
Description      1454
Quantity            0
InvoiceDate         0
Price               0
Customer ID    135080
Country             0
dtype: int64

* The Above checks for null values for each column and print the total null values in the end column vice.

* The below code fill the null values for the all the columns in the given dataset using for loop
* It checks for the column datatypes and calculate the mean for numerical and mode for the categorical columns to be filled in the dataset.'

In [409]:
# Check for null values
print("Null values before handling:")
print(data.isnull().sum())

# Handle null values
# For example, let's say we want to impute null values with the mean for numerical columns
# and with the mode for categorical columns
for col in data.columns:
    if data[col].dtype == 'object':  # Categorical columns
        mode_val = data[col].mode()[0]
        data[col].fillna(mode_val, inplace=True)
    else:  # Numerical columns
        mean_val = data[col].mean()
        data[col].fillna(mean_val, inplace=True)

# Check again for null values after handling
print("\nNull values after handling:")
print(data.isnull().sum())


Null values before handling:
Invoice             0
StockCode           0
Description      1454
Quantity            0
InvoiceDate         0
Price               0
Customer ID    135080
Country             0
dtype: int64

Null values after handling:
Invoice        0
StockCode      0
Description    0
Quantity       0
InvoiceDate    0
Price          0
Customer ID    0
Country        0
dtype: int64


C:\Users\devas\AppData\Local\Temp\ipykernel_20000\1866056267.py:11: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data[col].fillna(mode_val, inplace=True)
C:\Users\devas\AppData\Local\Temp\ipykernel_20000\1866056267.py:14: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, 

### Task B: 3 points

1. Remove rows with cancelled operation.
2. Drop duplicates
3. Remove rows where Quantity and Price are greater than 0.

In [410]:
# Remove rows with cancelled operation
#Invoice Number -> If this code starts with C, it means that the operation has been canceled.
data=data[data['Invoice'].str[0] != 'C']


In [411]:
#2. drop duplicates rows
data=data.drop_duplicates()


In [427]:
#3.Remove rows where Quantity and Price are not greater than 0.
data = data[(data['Quantity'] > 0) & (data['Price'] > 0)]


### Task C: 5 points

1. Check for no of unique values for stockcode and description.
2. Remove Duplicates for description and stockcode.
3. Remove rows where stockcode count is greater than 1 when aggregated by description.
4. Remove rows where description count is greater than 1 when aggregated by stockcode.

In [414]:
#1. Check for no of unique values for stockcode and description.

stockcode_count = data['StockCode'].nunique()
print("Unique value count for StockCodes:", stockcode_count)

# Number of unique values for 'Description'
description_count = data['Description'].nunique()
print("Unique value count for Descriptions:", description_count)


Unique value count for StockCodes: 3922
Unique value count for Descriptions: 4026


In [416]:
#2. Remove duplicates for 'Description' and 'StockCode' combo with 
data.drop_duplicates(subset=['Description', 'StockCode'], inplace=True)



In [418]:
# 3. Remove rows where stockcode count is greater than 1 when aggregated by description.
description_stockcode_counts = data.groupby('Description')['StockCode'].count()
duplicate_stockcodes = description_stockcode_counts[description_stockcode_counts > 1].index
data = data[~data['Description'].isin(duplicate_stockcodes)]

In [419]:
# 4. Remove rows where description count is greater than 1 when aggregated by stockcode.
stockcode_description_counts = data.groupby('StockCode')['Description'].count()
duplicate_descriptions = stockcode_description_counts[stockcode_description_counts > 1].index
data = data[~data['StockCode'].isin(duplicate_descriptions)]

* The above code agrregte on both description and stockcode and removes the extra description and stockcode rows keeping 0n0 to 0ne relation bestween them

### Task D: 2 points

1. Remove rows where stockcode contains Post that indicates the postage cost.
2. Consider rows for 'Germany' Only.

In [421]:
# Removing rows where stockcode contains 'POST'
data = data[~data['StockCode'].str.contains('POST', case=False)]


# Display sample of DataFrame after filtering
print("Sample of DataFrame after filtering out rows with 'POST' in StockCode:")
print(data.head())


Sample of DataFrame after filtering out rows with 'POST' in StockCode:
  Invoice StockCode                          Description  Quantity  \
2  536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3  536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4  536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
5  536365     22752         SET 7 BABUSHKA NESTING BOXES         2   
6  536365     21730    GLASS STAR FROSTED T-LIGHT HOLDER         6   

      InvoiceDate  Price  Customer ID         Country  
2  12/1/2010 8:26   2.75      17850.0  United Kingdom  
3  12/1/2010 8:26   3.39      17850.0  United Kingdom  
4  12/1/2010 8:26   3.39      17850.0  United Kingdom  
5  12/1/2010 8:26   7.65      17850.0  United Kingdom  
6  12/1/2010 8:26   4.25      17850.0  United Kingdom  


In [428]:
# Filter rows for 'Germany' only
data = data[data['Country'] == 'Germany']



# Display sample of DataFrame after filtering
print("Sample of DataFrame after filtering for 'Germany' only:")
display(data.head())


Sample of DataFrame after filtering for 'Germany' only:


Invoice StockCode                          Description  Quantity  \
1111  536527     84945   MULTI COLOUR SILVER T-LIGHT HOLDER        12   
1113  536527     22244           3 HOOK HANGER MAGIC GARDEN        12   
1115  536527     47421  ASSORTED COLOUR LIZARD SUCTION HOOK        24   
1116  536527     20712           JUMBO BAG WOODLAND ANIMALS        10   
1120  536527     22973         CHILDREN'S CIRCUS PARADE MUG        12   

          InvoiceDate  Price  Customer ID  Country  
1111  12/1/2010 13:04   0.85      12662.0  Germany  
1113  12/1/2010 13:04   1.95      12662.0  Germany  
1115  12/1/2010 13:04   0.42      12662.0  Germany  
1116  12/1/2010 13:04   1.95      12662.0  Germany  
1120  12/1/2010 13:04   1.65      12662.0  Germany

### Task E: 7 points

Transform the data having unique stockcode as column name and indication of the product in the each invoice as 0/1.

In [429]:
# Perform one-hot encoding with pivot table 
data_encoded = data.pivot_table(index='Invoice', columns='StockCode', aggfunc='size', fill_value=0)


display(data_encoded)


StockCode  20712  21084  21088  21096  21394  21424  22046  22244  22344  \
Invoice                                                                    
536527         1      0      0      0      0      0      0      1      0   
536983         0      0      0      0      0      0      0      0      0   
537197         0      0      0      0      0      0      0      0      0   
537212         0      0      0      0      0      0      0      0      1   
538174         0      0      0      0      0      0      0      0      0   
539405         0      0      0      0      0      0      0      0      0   
539446         0      0      0      0      0      1      0      0      0   
539946         0      0      0      0      1      0      0      0      0   
539957         0      0      0      0      0      0      1      0      0   
540458         0      1      1      1      0      0      0      0      0   
543121         0      0      0      0      0      0      0      0      0   
547821         0      0      0      0      0      0      0      0      0   
550137         0      0      0      0      0      0      0      0      0   
555282         0      0      0      0      0      0      0      0      0   
555383         0      0      0      0      0      0      0      0      0   
559298         0      0      0      0      0      0      0      0      0   
563950         0      0      0      0      0      0      0      0      0   
571739         0      0      0      0      0      0      0      0      0   

StockCode  22346  ...  23375  23702  44242B  47351B  47421  84569B  84849D  \
Invoice           ...                                                        
536527         0  ...      0      0       0       0      1       1       0   
536983         0  ...      0      0       0       0      0       0       1   
537197         0  ...      0      0       0       0      0       0       0   
537212         1  ...      0      0       0       0      0       0       0   
538174         0  ...      0      0       0       0      0       0       0   
539405         0  ...      0      0       0       0      0       0       0   
539446         0  ...      0      0       0       0      0       0       0   
539946         0  ...      0      0       0       0      0       0       0   
539957         0  ...      0      0       0       0      0       0       0   
540458         0  ...      0      0       0       1      0       0       0   
543121         0  ...      0      0       0       0      0       0       0   
547821         0  ...      0      0       0       0      0       0       0   
550137         0  ...      0      0       1       0      0       0       0   
555282         0  ...      0      0       0       0      0       0       0   
555383         0  ...      0      0       0       0      0       0       0   
559298         0  ...      0      0       0       0      0       0       0   
563950         0  ...      1      0       0       0      0       0       0   
571739         0  ...      0      1       0       0      0       0       0   

StockCode  84945  85205B  90098  
Invoice                          
536527         1       0      0  
536983         0       0      0  
537197         0       0      0  
537212         0       0      0  
538174         0       0      0  
539405         0       1      0  
539446         0       0      0  
539946         0       0      0  
539957         0       0      0  
540458         0       0      0  
543121         0       0      1  
547821         0       0      0  
550137         0       0      0  
555282         0       0      0  
555383         0       0      0  
559298         0       0      0  
563950         0       0      0  
571739         0       0      0  

[18 rows x 31 columns]

* performing the encoding for datset the between invoice and stockcode using pivot command

### Task F: 7 points

Apply apriori algorithm with 0.01 support

In [430]:
# implemting apriori algoritm with 0.01 support to find the ferquent item sets.
import pandas as pd
from mlxtend.frequent_patterns import apriori

frequent_itemsets_apriori = apriori(data_encoded, min_support=0.01, use_colnames=True)

C:\Users\devas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mlxtend\frequent_patterns\fpcommon.py:109: DeprecationWarning: DataFrames with non-bool types result in worse computationalperformance and their support might be discontinued in the future.Please use a DataFrame with bool type
  warnings.warn(


### Task G: 5 points

Print top 5 association rules for support metric and 0.01 threshold in accordance with support value.

In [431]:
# creating  the rules conditions for the frequent datasets

from mlxtend.frequent_patterns import association_rules

rules_apriori = association_rules(frequent_itemsets_apriori, metric="support", min_threshold=0.01)

rules_apriori.head()

antecedents consequents  antecedent support  consequent support   support  \
0     (20712)     (22244)            0.055556            0.055556  0.055556   
1     (22244)     (20712)            0.055556            0.055556  0.055556   
2     (20712)     (22973)            0.055556            0.055556  0.055556   
3     (22973)     (20712)            0.055556            0.055556  0.055556   
4     (47421)     (20712)            0.055556            0.055556  0.055556   

   confidence  lift  leverage  conviction  zhangs_metric  
0         1.0  18.0  0.052469         inf            1.0  
1         1.0  18.0  0.052469         inf            1.0  
2         1.0  18.0  0.052469         inf            1.0  
3         1.0  18.0  0.052469         inf            1.0  
4         1.0  18.0  0.052469         inf            1.0

### Task H: 3 points

Store the association rules from top to bottom with respect to lift parameter and store it in a variable.

In [432]:
# Sorting the rules using wrt lift parameter

rules_lift_sorted = rules_apriori.sort_values(by='lift', ascending=False)

Task I: 10 points

Understand this,
- antecedent support: If X is called antecendent, 'antecedent support' computes the proportion of transactions that contain the antecedent X.
- consequent support: If Y is called consequent, 'consequent support' computes the proportion of transactions that contain the antecedent Y.
- support: 'support' computes the proportion of transactions that contain the antecedent X and Y.
- confidence: Probability of buying Y when X is bought.
- lift: Represents how many times the probability of getting Y increases when X is received.

Define a function which takes the (sorted values from the association rules,a product name) and returns top recommendation list for that product name.

Tip: the recommendated item are Consequents by matching the product antecedents as product to search for.

In [341]:
# defining a function to provide recommendations with given product name and sorted list of rules
# default is 5 recommendations

def list_recommendation(rules_sorted, item_name, top_number=5):
    recommendations = rules_sorted[rules_sorted['antecedents'].apply(lambda x: item_name in x)]
    return recommendations.head(top_number)['consequents'].tolist()


* The above function checks for the given product name in antecedents columns and gets the resultds from the copnsequent column if the given column is present.

### Task J: 5 points

Print top 5 reccomendated items for product '22492'.

In [342]:
# getting the 5 top recommendations for the product using predefined fuction

top5=list_recommendation(rules_lift_sorted,'22492')
print("The Top 5 reccomendend items for the product '22492':")
display(top5)

The Top 5 reccomendend items for the product '22492':


[]

### Task K: 5 points

Print top 5 reccomendated items for product '22492' with 0.02 support.

In [344]:
# getting the 5 top recommendations for the product using predefined fuction but with support of 0.02

# applying all the steps for apriori with 0.02 support 
frequent_itemsets_apriori2 = apriori(data_encoded, min_support=0.02, use_colnames=True)
rules_apriori2 = association_rules(frequent_itemsets_apriori2, metric="support", min_threshold=0.01)
rules_lift_sorted2 = rules_apriori2.sort_values(by='lift', ascending=False)


top_list=list_recommendation(rules_lift_sorted2,'22492')
print("The Top 5 reccomendend items for the product '22492' with 0.02 support:")
display(top_list)


The Top 5 reccomendend items for the product '22492' with 0.02 support:


C:\Users\devas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mlxtend\frequent_patterns\fpcommon.py:109: DeprecationWarning: DataFrames with non-bool types result in worse computationalperformance and their support might be discontinued in the future.Please use a DataFrame with bool type
  warnings.warn(


[]

getting the 5 top recommendations for the product using predefined fuction but with support of 0.02

 applying all the steps for apriori with 0.02 support 

### Task L: 10 points

Perform task F through K except task I using FP-Growth algorithm.

In [349]:
# Creating the frequent datasets with fpgrowth algorithm
# and sorting the rules with respoect to lift 

from mlxtend.frequent_patterns import fpgrowth
frequent_itemsets_fp = fpgrowth(data_encoded, min_support=0.01, use_colnames=True)

rules_fp = association_rules(frequent_itemsets_fp, metric="support", min_threshold=0.01)
display(rules_fp.head())
rules_sorted_fp = rules_fp.sort_values(by='lift', ascending=False)

C:\Users\devas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mlxtend\frequent_patterns\fpcommon.py:109: DeprecationWarning: DataFrames with non-bool types result in worse computationalperformance and their support might be discontinued in the future.Please use a DataFrame with bool type
  warnings.warn(


antecedents consequents  antecedent support  consequent support   support  \
0     (22244)     (22973)            0.055556            0.055556  0.055556   
1     (22973)     (22244)            0.055556            0.055556  0.055556   
2     (47421)     (22973)            0.055556            0.055556  0.055556   
3     (22973)     (47421)            0.055556            0.055556  0.055556   
4     (47421)     (22244)            0.055556            0.055556  0.055556   

   confidence  lift  leverage  conviction  zhangs_metric  
0         1.0  18.0  0.052469         inf            1.0  
1         1.0  18.0  0.052469         inf            1.0  
2         1.0  18.0  0.052469         inf            1.0  
3         1.0  18.0  0.052469         inf            1.0  
4         1.0  18.0  0.052469         inf            1.0

In [350]:
# 5 top recommendations with fp growth and 0.01 support

top_list_fp=list_recommendation(rules_sorted_fp,'22492')
print("\nThe Top 5 reccomendend items for the product '22492' using F-P growth:")
print(top_list_fp)


The Top 5 reccomendend items for the product '22492':
[]


In [433]:
# 5 top recommendations with fp growth with 0.02 support

frequent_itemsets_fp2 = fpgrowth(data_encoded, min_support=0.02, use_colnames=True)

rules_fp2 = association_rules(frequent_itemsets_fp, metric="support", min_threshold=0.01)
rules_sorted_fp2 = rules_fp.sort_values(by='lift', ascending=False)



C:\Users\devas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mlxtend\frequent_patterns\fpcommon.py:109: DeprecationWarning: DataFrames with non-bool types result in worse computationalperformance and their support might be discontinued in the future.Please use a DataFrame with bool type
  warnings.warn(


In [353]:


top_list_fp2=list_recommendation(rules_sorted_fp2,'22492')
print("The Top 5 reccomendend items for the product '22492' using F-P growth: with 0.02 support:")
display(top_list_fp2)

The Top 5 reccomendend items for the product '22492' using F-P growth: with 0.02 support:


[]

### Task M: 8 points

Print Top 3 recommendations for below products using Apriori with 0.015 support
- SET OF 4 PANTRY JELLY MOULDS
- 3 HOOK HANGER MAGIC GARDEN
- PACK OF 20 NAPKINS PANTRY DESIGN

In [359]:
# providing the top 3 reccomendation for the given description products using aporiori and 0.015 support

raw_data = pd.read_csv("Data_batch2.csv", encoding= 'unicode_escape')


for name in ['SET OF 4 PANTRY JELLY MOULDS', '3 HOOK HANGER MAGIC GARDEN' , 'PACK OF 20 NAPKINS PANTRY DESIGN']:
    scode=raw_data[raw_data["Description"]==name].iloc[0]["StockCode"]
    frequent_itemsets_apriori_name = apriori(data_encoded, min_support=0.015, use_colnames=True)
    rules_apriori_name = association_rules(frequent_itemsets_apriori_name, metric="support", min_threshold=0.01)
    rules_lift_sorted2_name = rules_apriori_name.sort_values(by='lift', ascending=False)


    top_list_name=list_recommendation(rules_lift_sorted2,scode,3)
    print(f"\nThe Top 3 reccomendend items for the product {name} with 0.015 support using apriori:")
    display(top_list)

22993

The Top 3 reccomendend items for the product SET OF 4 PANTRY JELLY MOULDS with 0.015 support using apriori:


C:\Users\devas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mlxtend\frequent_patterns\fpcommon.py:109: DeprecationWarning: DataFrames with non-bool types result in worse computationalperformance and their support might be discontinued in the future.Please use a DataFrame with bool type
  warnings.warn(


[]

22244

The Top 3 reccomendend items for the product 3 HOOK HANGER MAGIC GARDEN with 0.015 support using apriori:


C:\Users\devas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mlxtend\frequent_patterns\fpcommon.py:109: DeprecationWarning: DataFrames with non-bool types result in worse computationalperformance and their support might be discontinued in the future.Please use a DataFrame with bool type
  warnings.warn(


[]

22907

The Top 3 reccomendend items for the product PACK OF 20 NAPKINS PANTRY DESIGN with 0.015 support using apriori:


C:\Users\devas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mlxtend\frequent_patterns\fpcommon.py:109: DeprecationWarning: DataFrames with non-bool types result in worse computationalperformance and their support might be discontinued in the future.Please use a DataFrame with bool type
  warnings.warn(


[]

### Task N: 8 points

Print Top 3 recommendations for below products using FP-Growth algorithm with 0.015 support
- COFFEE MUG PEARS  DESIGN
- PLASTERS IN TIN WOODLAND ANIMALS
- PACK OF 20 NAPKINS RED APPLES

In [426]:
# providing the top 3 reccomendation for the given description products using fp growth and 0.015 support
for name in ['COFFEE MUG PEARS  DESIGN', 'PLASTERS IN TIN WOODLAND ANIMALS', 'PACK OF 20 NAPKINS RED APPLES']:
    scode=raw_data[raw_data["Description"]==name].iloc[0]["StockCode"]
    frequent_itemsets_fp_name = fpgrowth(data_encoded, min_support=0.015, use_colnames=True)
    rules_fp_name = association_rules(frequent_itemsets_fp_name, metric="support", min_threshold=0.01)
    rules_lift_fp_name = rules_fp_name.sort_values(by='lift', ascending=False)


    top_list_fp_name=list_recommendation(rules_lift_fp_name,scode,3)
    print(f"\nThe Top 3 reccomendend items for the product {name} with 0.015 support using FP-Growth:")
    display(top_list_fp_name)


The Top 3 reccomendend items for the product COFFEE MUG PEARS  DESIGN with 0.015 support using FP-Growth:


C:\Users\devas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mlxtend\frequent_patterns\fpcommon.py:109: DeprecationWarning: DataFrames with non-bool types result in worse computationalperformance and their support might be discontinued in the future.Please use a DataFrame with bool type
  warnings.warn(


[]


The Top 3 reccomendend items for the product PLASTERS IN TIN WOODLAND ANIMALS with 0.015 support using FP-Growth:


C:\Users\devas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mlxtend\frequent_patterns\fpcommon.py:109: DeprecationWarning: DataFrames with non-bool types result in worse computationalperformance and their support might be discontinued in the future.Please use a DataFrame with bool type
  warnings.warn(


[]


The Top 3 reccomendend items for the product PACK OF 20 NAPKINS RED APPLES with 0.015 support using FP-Growth:


C:\Users\devas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\mlxtend\frequent_patterns\fpcommon.py:109: DeprecationWarning: DataFrames with non-bool types result in worse computationalperformance and their support might be discontinued in the future.Please use a DataFrame with bool type
  warnings.warn(


[]

### Task O: 20 points

Submit a document showing your solution for below problem.

For the data on apriori image in the folder, apply apriori algorithm by

- Computing the support for each individual item
- Use 7 as the minimum support.
- Generate Association Rules and compute confidence 
- use min confidence as 85%.

### Programming Assignment Details

1. If using any resource (books, internet), please make sure that you cite it within that cell.
2. Do not rename the dataset_files.

###  Submission details
Fill your name and ID in the jupyter notebook for each group member in the following format:

1. First Student Name and ID: ABC 1001XXXXXX
2. Second Student Name and ID: DEF 1002XXXXXX
3. Third Student Name and ID: GHI 1003XXXXXX

Name your submission files:

yourLastName_Last4digitsofyourID.ipynb

EG: abc_1234_def_5678_xyz3819_python.ipynb

### NOTE: Only one team member will submit the file.